# Estadistica

## Unidad 1
1. Teorema de Bayes.
2. Variables aleatorias
3. Función de distribución.
4. Distribuciones bivariadas, marginales y condicionales

In [99]:
def IC_norm_sigma_conocida(media_muestral:float, varianza: float, n:int, confianza: int ) -> tuple:
    """
    Parametros(
    media_muestral:float, 
    varianza: float, 
    n:int, 
    confianza: int 
    )
    """
    alpha = 1 - confianza/100
    alpha_2 = alpha/2
    Z_alpha_2 = norm.ppf(1-alpha_2)
    print(Z_alpha_2)
    limite_1 = media_muestral - Z_alpha_2*((varianza**(0.5))/((n)**(0.5)))
    limite_2 = media_muestral + Z_alpha_2*((varianza**(0.5))/((n)**(0.5)))
    if (limite_1 < limite_2):
        print(limite_2 - limite_1)
        return (limite_1, limite_2)
    
    else :
        print(limite_1 - limite_2)
        return (limite_2, limite_1)

def IC_norm_sigma_desconocida(media_muestral:float, varianza_muestral: float, n:int, confianza: int ) -> tuple:
    """
    Parametros(
    media_muestral:float, 
    varianza_muestral: float, 
    n:int, 
    confianza: int 
    )
    return: tupla con limites del intervalo de confianza
    """
    alpha = 1 - confianza/100
    alpha_2 = alpha/2
    t_alpha_2 = t.ppf(1 - alpha_2, n-1)
    limite_1 = media_muestral - t_alpha_2*((varianza_muestral**(0.5))/((n)**(0.5)))
    limite_2 = media_muestral + t_alpha_2*((varianza_muestral**(0.5))/((n)**(0.5)))
    if (limite_1 < limite_2):
        return (limite_1, limite_2)
    else :
        return (limite_2, limite_1)
    
def IC_no_norm_sigma_conocida(media_muestral:float, varianza_muestral: float, n:int, confianza: int ) -> tuple:
    """
    Parametros(
    media_muestral:float, 
    varianza_muestral: float, 
    n:int, 
    confianza: int 
    )
    """
    alpha = 1 - confianza/100
    alpha_2 = alpha/2
    Z_alpha_2 = norm.ppf(1-alpha_2)
    print(Z_alpha_2)
    limite_1 = media_muestral - Z_alpha_2*((varianza_muestral**(0.5))/((n)**(0.5)))
    limite_2 = media_muestral + Z_alpha_2*((varianza_muestral**(0.5))/((n)**(0.5)))
    if (limite_1 < limite_2):
        print(limite_2 - limite_1)
        return (limite_1, limite_2)
    
    else :
        print(limite_1 - limite_2)
        return (limite_2, limite_1)
    
    
def IC_x_y_norm_sigma_desconocida(
                                  media_muestral_1:float, 
                                  media_muestral_2:float,
                                  varianza_muestral_1:float,
                                  varianza_muestral_2: float,
                                  n:int,
                                  m:int,
                                  confianza
                                 ):
    alpha = 1 - confianza/100
    alpha_2 = alpha/2
    numerador_v = ((( ((varianza_muestral_1)**2)/n + ((varianza_muestral_2)**2)/m )**2)*(n-1)*(m-1))
    denominador_v = ((((varianza_muestral_1)**2)/n)**2)*(m-1) + ((((varianza_muestral_2)**2)/m)**2)*(n-1)
    v = numerador_v / denominador_v
    print("v",v)
    t_alpha_2 = t.ppf(1 - alpha_2, round(v))
    radical = ( (((varianza_muestral_1)**2)/n) + (((varianza_muestral_2)**2)/m) )**0.5
    limite_1 = media_muestral_1 - media_muestral_2 - t_alpha_2*radical
    limite_2 = media_muestral_1 - media_muestral_2 + t_alpha_2*radical
    if (limite_1 < limite_2):
        return (limite_1, limite_2)
    else :
        return (limite_2, limite_1)
    
    
def IC_x_y_no_norm_sigma_desconocida(
                                  media_muestral_1:float, 
                                  media_muestral_2:float,
                                  varianza_muestral_1:float,
                                  varianza_muestral_2: float,
                                  n:int,
                                  m:int,
                                  confianza
                                 ):
    alpha = 1 - confianza/100
    alpha_2 = alpha/2
    numerador_v = ((( ((varianza_muestral_1)**2)/n + ((varianza_muestral_2)**2)/m )**2)*(n-1)*(m-1))
    denominador_v = ((((varianza_muestral_1)**2)/n)**2)*(m-1) + ((((varianza_muestral_2)**2)/m)**2)*(n-1)
    t_alpha_2 = norm.ppf(1 - alpha_2)
    radical = ( (((varianza_muestral_1)**2)/n) + (((varianza_muestral_2)**2)/m) )**0.5
    limite_1 = media_muestral_1 - media_muestral_2 - t_alpha_2*radical
    limite_2 = media_muestral_1 - media_muestral_2 + t_alpha_2*radical
    if (limite_1 < limite_2):
        return (limite_1, limite_2)
    else :
        return (limite_2, limite_1)
    

## Ejercicios para taller 2

Intervalos de confianza 
1. IC para $\mu$
    1. Mirar so la m.a 
    2. Normal
        1. $\sigma^2$ conocida
        2. $\sigma^2$ desconocida

In [19]:
import pandas as pd
from  scipy.stats import norm, t
import numpy as np
import sympy as sp

I.C para $\mu$
$
\begin{align}

\end{align}
$

# Intervalos de Confianza Para la Media

<img src="./Imagenes/estadistica/ejercicio1.png">

In [3]:
# Muestra
list_1 = [23, 39, 40, 41, 43, 47, 51, 58, 63, 66, 67, 69, 72]

In [4]:
# Calculando la media y varianza muestral y definimos la cantidad de elementos.
media_muestral = np.mean(list_1)
varianza_muestral = np.var(list_1, ddof=1)
n = len(list_1)

In [5]:
# Cálculamos la significancia que es alpha y alpha  
confianza = 0.95 # 
alpha     = 1 - confianza
alpha_2   = alpha/2

In [6]:
# Calculando la imagen la del perceptil con respecto a la distribución
t_alpha_2_n_1 = t.ppf(1-alpha_2, n-1)
t_alpha_2_n_1

2.1788128296634177

In [7]:
# Se cálcula el intervalo:
(media_muestral + t_alpha_2_n_1*((varianza_muestral**(0.5))/((n)**(0.5))) , media_muestral - t_alpha_2_n_1*((varianza_muestral**(0.5))/((n)**(0.5))))

(61.20798876305008, 43.253549698488385)

<h4 style="color: blue">
Conclusión: Con un 95% de confianza se puede afirmar que el peso promedio de los animales sanos se encuentra entre 43.2535 y 61.20798 respecto a la unidad de peso.
</h4>

<img src="./Imagenes/estadistica/ejercicio2.png">

In [8]:
IC_norm_sigma_conocida(40.9, 16, 16, 98)

2.3263478740408408
4.652695748081683


(38.57365212595916, 43.22634787404084)

In [9]:
"Ejericico 3 restar los intervalos menos "
"(2*4*2.3263)**2 < n"

'(2*4*2.3263)**2 < n'

In [10]:
IC_norm_sigma_conocida(40.9, 16, 347, 98)

2.3263478740408408
0.999079177754254


(40.40046041112287, 41.399539588877126)

In [11]:
#Ejercicio 4
IC_norm_sigma_desconocida(30.2, (3.1)**2, 18, 95)

(28.658406267789847, 31.741593732210152)

In [12]:
# intervalo de confianza para la diferencia de medias

In [13]:
#Ejercicio 1 de diferencia de medias
IC_x_y_norm_sigma_desconocida(548,517,57,52,15,15,95)

v 27.767268908886958


(-9.807368808320518, 71.80736880832052)

$\mu_1 - \mu_2 \in (-9.822783212529636, 71.82278321252963)$, en particular $\mu_1 - \mu_2 = 0$, por lo cual $\mu_1 = \mu_2$

In [14]:
#Ejercicio 2 de diferencia de medias

IC_x_y_no_norm_sigma_desconocida(548,517,57,52,45,35,95)

(7.0389178094078275, 54.96108219059217)

$\mu_1 - \mu_2 \in (7.0389178094078275, 54.96108219059217)$, existe un $\beta>0 \in (7.0389178094078275, 54.96108219059217)$  tal que $\mu_1 - \mu_2 = \beta>0$, por lo cual $\mu_1 > \mu_2$.

In [15]:
IC_x_y_norm_sigma_desconocida(35.22, 31.56, 4.944, 4.475, 9, 9, 95)

v 15.843649081977317


(-1.0521892085864009, 8.372189208586402)

$\mu_1 - \mu_2 \in (-1.055971345970872, 8.375971345970871)$, en particular $\mu_1 - \mu_2 = 0$, por lo cual $\mu_1 = \mu_2$

#Distribuciones Conjuntas 
<img src="./Imagenes/estadistica/ejercicio1Conjuntas.png">

1. Sean $X$, $Y$ variabnles aleatorias continuas con f.d.p conjuna dada por: 


$
\begin{align}
f(x,y) =\Biggl\{ 
    \begin{split} 
     3x,&\text{ } 0<y<x<1,\\
      0,&\text{ otros casos }
    \end{split}
\end{align}
$

La distribución condicional de $Y$ dado $X=\frac{1}{2}$ 


$
\begin{align}
f_{Y|x=1/2}(y,1/2) = \frac{f(1/2,y)}{f_{X}(1/2)} 
&= \frac{3(1/2)}{\int_{-\infty}^{\infty} f(1/2,y)dy}\\
&= \frac{3(1/2)}{\int_{0}^{1/2} 3(1/2)dy} \\
&= \frac{3(1/2)}{3(1/2) [y]|_{0}^{1/2}} \\
&= \frac{1}{1/2} \\
&= 2 \\
\end{align}
$

$
\begin{align}
E[X|Y=1/2] = \int_{-\infty}^{\infty} xf(x|y=1/2)dx
&=\int_{-\infty}^{\infty} x\frac{f(x,y=1/2)}{f_{Y}(1/2)}dx,\\
&=\int_{0}^{1} x\frac{3x}{f_{Y}(1/2)}dx,\\
&=\int_{0}^{1} x\frac{3x}{\int_{0}^{1}3x dx}dx,\\
&=\int_{0}^{1} \frac{3x^2}{3\int_{0}^{1}x dx}dx,\\
&=2\int_{0}^{1} \frac{x^2}{x^2|_{0}^{1}}dx,\\
&=\frac{2}{3}
\end{align}
$


$
\begin{align}
Cov[X,Y] = E[XY] - E[X]E[Y] &= \int_{-\infty}^{\infty}\int_{-\infty}^{\infty} xyf(x,y)dydx - \Biggl(\int_{-\infty}^{\infty}\int_{-\infty}^{\infty}xf(x,y)dydx \Biggl)\Biggl( \int_{-\infty}^{\infty}\int_{-\infty}^{\infty}yf(x,y)dydx \Biggl)\\
&= \int_{0}^{1}\int_{0}^{x} xy3xdydx - \Biggl(\int_{0}^{1}\int_{0}^{x}xf(x,y)dydx \Biggl)\Biggl( \int_{0}^{1}\int_{0}^{x}yf(x,y)dydx \Biggl)\\
\end{align}
$


$
P(X\leq \frac{\pi}{4}) = \int_{0}^{\pi/2}\int_{0}^{\pi/4} \frac{1}{\pi} (cos(x) + cos(y)) dxdy
$


In [30]:
x,y = sp.symbols('x y')
f = sp.cos(x) + sp.cos(y)
resultado = 1/sp.pi *(sp.integrate(sp.integrate(f, (x, 0, sp.pi/4)), (y, 0, sp.pi/2) ))
resultado

(pi/4 + sqrt(2)*pi/4)/pi

Ejercicio 3 

$
1 =\sum_{y=1}^{3}\sum_{x=1}^{3} kxy \equiv  \frac{1}{\sum_{y=1}^{3}\sum_{x=1}^{3} xy} =k
$



In [46]:
expresion  = 4*x*y 

suma_doble = sp.summation(sp.summation(expresion, (x, 1,3)), (y,1,3))
suma_doble

k = 1/suma_doble
k

1/144

$
cov[X,Y] = E[XY] - E[X]E[Y] =\sum_{y=1}^{3}\sum_{x=1}^{3} (1/144)(xy)^{2} - \Biggl(\sum_{y=1}^{3}\sum_{x=1}^{3} (1/144)x^2y\biggl)\biggl( \sum_{y=1}^{3}\sum_{x=1}^{3} (1/144)y^2x \biggl)
$

In [58]:
E_XY = (1/144)*(x*y)**2 

E_XY = sp.summation(sp.summation(E_XY, (x, 1,3)), (y,1,3))
XY

E_X = (1/144)*((x)**2)*y
E_X = sp.summation(sp.summation(E_X, (x, 1,3)), (y,1,3))

E_Y = (1/144)*((y)**2)*x
E_Y = sp.summation(sp.summation(E_Y, (x, 1,3)), (y,1,3))

cov_XY = E_XY - E_X*E_Y
print("cov[XY] =",cov_XY)
print("E[x] =", E_X)

cov[XY] = 1.02083333333333
E[x] = 0.583333333333333


In [84]:
dataset  = pd.read_csv('data/Dioxido.txt', sep='\t')
dataset.head()

dioxidocarbono   x1   x2     x3     x4  x5
0             107  188  2.0  27.44  28.57   5
1             107  188  2.0  27.44  28.57   0
2             123  188  2.0  25.62  24.77   2
3             123  188  2.0  25.62  24.77   2
4             135  156  2.5  24.00  21.90   5

In [89]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50 entries, 0 to 49
Data columns (total 6 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   dioxidocarbono  50 non-null     int64  
 1   x1              50 non-null     int64  
 2   x2              50 non-null     float64
 3   x3              50 non-null     float64
 4   x4              50 non-null     float64
 5   x5              50 non-null     int64  
dtypes: float64(3), int64(3)
memory usage: 2.5 KB


In [92]:
dataset.shape

(50, 6)

In [113]:
medias_muestrales = dataset.mean()
varianzas_muestrales = dataset.var(ddof=1)
desviacion_muestrales = varianzas_muestrales**0.5
n=50
alpha = 1 - 95/100
alpha_2 = alpha/2

z_alpha_2 = norm.ppf(1 - alpha_2)
#IC_no_norm_sigma_conocida(medias_muestrales, varianzas_muestrales,50, 95)
IC =pd.concat([medias_muestrales - z_alpha_2* desviacion_muestrales/((n)**0.5), medias_muestrales + z_alpha_2* desviacion_muestrales/((n)**0.5)], axis=1)
#IC_no_norm_sigma_conocida()
IC

0           1
dioxidocarbono  145.400919  154.599081
x1              122.543596  137.336404
x2                1.532121    1.699879
x3               17.240712   19.131288
x4               24.140226   25.344574
x5               12.482538   24.917462